Implement PublicationOnboardingStateNotice component.

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #8838 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

hussain-t commented on assets/js/modules/reader-revenue-manager/components/common/PublicationOnboardingStateNotice.js:
```diff
@@ -0,0 +1,99 @@
+		// Get the service URL from the store.
+		return (
+			<Link href={ serviceURL } external inverse>
+				{ __( 'Complete publication setup', 'google-site-kit' ) }
```
The CTA text should be `Check publication status` for the `PENDING_VERIFICATION` onboarding state. Please take a look at the [Figma design](https://www.figma.com/design/f2AFGUqfH8rqi9BpJAwhHi/Reader-Revenue-Manager?node-id=213-3192&m=dev).

hussain-t commented on assets/js/modules/reader-revenue-manager/components/common/PublicationOnboardingStateNotice.js:
```diff
@@ -0,0 +1,99 @@
+	};
+
+	const noticeText =
+		PENDING_VERIFICATION === onboardingstate
+			? __(
+					'Your publication is still awaiting review. you can check its status in Reader Revenue Manager.',
+					'google-site-kit'
+			  )
+			: __(
+					'Your publication requires further setup in Reader Revenue Manager.',
+					'google-site-kit'
+			  );
```
As per the Figma design, the copies are inconsistent, with a period at the end. I have [raised](https://www.figma.com/design/f2AFGUqfH8rqi9BpJAwhHi?node-id=190-2315&m=dev#884752962) it with @sigal-teller. Let's wait for the response and update them.

hussain-t commented on assets/js/modules/reader-revenue-manager/components/common/PublicationOnboardingStateNotice.test.js:
```diff
@@ -0,0 +1,159 @@
+		expect( container ).toBeEmptyDOMElement();
+	} );
+
+	it( 'should have "googlesitekit-publication-onboarding-state-notice" class in the container', async () => {
```
IMO, asserting the text should be sufficient. As for the classes, we could nest it within the text assertion tests. I'll leave it to you.

hussain-t reviewed (changes requested):
Thanks, @ankitrox. It's a good start. However, I've left some comments. Please take a look at them.

ankitrox commented:
Thanks for reviewing this @hussain-t .

I have addressed all comments except the one you want to clarify with Sigal. Over to you for re-review.

Thank you

hussain-t commented on assets/js/modules/reader-revenue-manager/components/common/PublicationOnboardingStateNotice.test.js:
```diff
@@ -0,0 +1,140 @@
+		expect( container ).toBeEmptyDOMElement();
+	} );
+
+	it( 'should render the "Check publication status" CTA with correct link', async () => {
```
It is also better to test the `Complete publication setup` state. Overall, the tests can be combined and simplified. I will push the changes to avoid another iteration.

hussain-t reviewed (approved):
Thanks, @ankitrox. LGTM 👍 

As you will have seen my comments, I've made a couple of changes.

techanvil reviewed (changes requested):
Hi @ankitrox, this is generally OK, but needs a couple of tweaks and also a change to the approach for QA.

ankitrox commented:
Thanks @techanvil . I've added the story for the component and updated QAB as suggested.

Over to you for another review.

techanvil commented on assets/js/modules/reader-revenue-manager/components/common/PublicationOnboardingStateNotice.stories.js:
```diff
@@ -0,0 +1,87 @@
+export default {
+	title: 'Modules/ReaderRevenueManager/Common/PublicationOnboardingStateNotice',
+	decorators: [
+		( Story ) => {
+			const registry = createTestRegistry();
+
+			return (
+				<WithTestRegistry registry={ registry }>
+					<Story />
+				</WithTestRegistry>
+			);
+		},
```
We don't need both `WithTestRegistry` and `WithRegistrySetup` wrapping the story, just one, and preferably `WithRegistrySetup` as this avoids the need to create the registry in the story.

I'd suggest moving the `WithRegistrySetup` here from the `Template` component:

```suggestion
		( Story, { args } ) => {
			return (
				<WithRegistrySetup func={ args.setupRegistry }>
					<Story />
				</WithRegistrySetup>
			);
		},
```

techanvil commented on assets/js/modules/reader-revenue-manager/components/common/PublicationOnboardingStateNotice.stories.js:
```diff
@@ -0,0 +1,87 @@
+const { PENDING_VERIFICATION, ONBOARDING_ACTION_REQUIRED } =
+	PUBLICATION_ONBOARDING_STATES;
+
+function Template( { setupRegistry = async () => {}, ...args } ) {
+	const setupRegistryCallback = async ( registry ) => {
+		await setupRegistry( registry );
+	};
+
+	return (
+		<WithRegistrySetup func={ setupRegistryCallback }>
+			<PublicationOnboardingStateNotice { ...args } />
+		</WithRegistrySetup>
+	);
+}
```
As discussed below, we should move the `WithRegistrySetup` wrapper into the decorator. This `Template` component can be simplified further, as `PublicationOnboardingStateNotice` doesn't have any props:

```suggestion
function Template() {
	return <PublicationOnboardingStateNotice />;
}
```

techanvil reviewed (changes requested):
Thanks @ankitrox. This needs another iteration, please see my comments.

techanvil commented on assets/sass/components/reader-revenue-manager/_index.scss:
```diff
@@ -1,6 +1,9 @@
 /**
+<<<<<<< HEAD
+=======
  * Reader Revenue Manager styles.
  *
+>>>>>>> develop
```
@ankitrox looks like some unwanted merge artifacts here.

ankitrox commented:
Thank you @techanvil.

I've addressed the comments and pushed the changes. Assigning this to you for one more review.

techanvil commented on assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-publication-onboarding-state-notice.scss:
```diff
@@ -0,0 +1,19 @@
+.googlesitekit-publication-onboarding-state-notice.googlesitekit-settings-notice--warning {
+	align-items: center;
+	color: $c-utility-on-warning-container;
+	max-width: none;
+
+	& .googlesitekit-cta-link.googlesitekit-cta-link--inverse {
+		background-color: $c-utility-on-warning-container;
+		border-radius: 100px;
+		padding: 6px 16px;
+	}
+
+	@media (max-width: $bp-tablet) {
+		align-items: flex-start;
+		flex-direction: column;
+		font-size: 14px;
+		gap: 10px;
+		line-height: 20px;
```
Hey @ankitrox, sorry if it wasn't so clear in my [comment](https://github.com/google/site-kit-wp/pull/9037/files/b3fe30cdea7507b8790b5284b653b3084642ba0c..323eb57f3319f7dc1c91f8ab7d35e6e73a4ae4e4#r1697201342), but the typography should be `label/medium` for all viewports to match Figma.

Also, we need to apply a gap between the text and the button, and also prevent line-wrapping in the button, to avoid these scenarios:

![image](https://github.com/user-attachments/assets/333b8174-033b-4e96-85a6-9b8908e31e36)

![image](https://github.com/user-attachments/assets/9fbda5e1-f2dd-4891-a461-db2a17b86a0e)

```suggestion
	font-size: $fs-label-md;
	gap: 10px;
	line-height: $lh-label-md;
	max-width: none;

	.googlesitekit-settings-notice__icon {
		margin-right: 0;
	}

	& .googlesitekit-cta-link.googlesitekit-cta-link--inverse {
		background-color: $c-utility-on-warning-container;
		border-radius: 100px;
		padding: 6px 16px;
		white-space: nowrap;
	}

	@media (max-width: $bp-tablet) {
		align-items: flex-start;
		flex-direction: column;
```

techanvil reviewed (approved):
LGTM, nice one @ankitrox!
